common/scriptcore/stubs/game.lua

```diff
@@ -0,0 +1,734 @@
+
+--- Functions in this module are used to acquire objects for further
+--- manipulation, given various kinds of identifying information. Functions are
+--- overloaded so that a given object can be identified in several ways. 
+---
+--- !doctype table
+--- @class find
+find = {}
+
+--- 
+--- @param player_id int The unique ID of the player to search for
+--- @return Player player The player with the given ID if they exist, or nil
+function find.player(player_id) end
+
+--- 
+--- @param player Player The player who owns the city, or nil for any player
+--- @param city_id int The unique ID for the city to search for
+--- @return City city The city if it exists and belongs to player, or nil
+function find.city(player, city_id) end
+
+--- 
+--- @param player Player The player who owns the unit, or nil for any player
+--- @param unit_id int The unique ID for the unit to search for
+--- @return Unit unit The unit if it exists and belongs to player, or nil
+function find.unit(player, unit_id) end
+
+--- Looks for an existing unit that can transport unit_type at tile. Intended
+--- to be used with create_unit_full, to spawn it directly into the hold of the
+--- transport.
+---
+--- @param player Player The player who owns the transport unit
+--- @param unit_type Unit_Type The unit type to search for a transport for
+--- @param tile Tile The tile to search in
+--- @return Unit unit The transport unit found, or nil
+function find.transport_unit(player, unit_type, tile) end
+
+--- Finds a tile based on natural coordinates.
+--- @param nat_x int The natural x-coordinate.
+--- @param nat_y int The natural y-coordinate.
+--- @return Tile tile The found tile.
+function find.tile(nat_x, nat_y) end
+
+--- Finds a tile by its index.
+--- @param tindex int The index of the tile.
+--- @return Tile tile The found tile.
+function find.tile(tindex) end
+
+--- Finds a government by its name.
+--- @param name string The ruleset name of the government.
+--- @return Government government The found government.
+function find.government(name) end
+
+--- Finds a government by its ID.
+--- @param government_id int The ID of the government.
+--- @return Government government The found government.
+function find.government(government_id) end
+
+--- Finds a nation type by its name.
+--- @param name string The ruleset name of the nation type.
+--- @return Nation_Type nation The found nation type.
+function find.nation_type(name) end
+
+--- Finds a nation type by its ID.
+--- @param nation_type_id int The ID of the nation type.
+--- @return Nation_Type nation The found nation type.
+function find.nation_type(nation_type_id) end
+
+--- Finds an action by its rule name.
+--- @param name string The ruleset name of the action.
+--- @return Action action The found action.
+function find.action(name) end
+
+--- Finds an action by its ID. For use in loops etc. 
+---
+--- .. attention::
+---    Warning: An action's ID may change after saving and reloading the game.
```

Comment:
I wonder if we need both an attention and a warning. There are a few places other than here that do the same thing.